modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3RepositoryPlugin.java

```diff
@@ -54,9 +63,26 @@ public class S3RepositoryPlugin extends Plugin implements RepositoryPlugin, Relo
             try {
                 // kick jackson to do some static caching of declared members info
                 Jackson.jsonNodeOf("{}");
-                // ClientConfiguration clinit has some classloader problems
-                // TODO: fix that
-                Class.forName("com.amazonaws.ClientConfiguration");
+
+                // some AWS classes emit noisy WARN logs so we hide them by setting logger to ERROR level
+                final Map<Logger, Level> awsConfigLoggers = List.of(
+                    AwsSdkMetrics.class,
+                    BasicProfileConfigFileLoader.class,
+                    SdkMBeanRegistrySupport.class,
+                    UseArnRegionResolver.class
+                ).stream().map(LogManager::getLogger).collect(Collectors.toUnmodifiableMap(logger -> logger, Logger::getLevel));
+                try {
+                    awsConfigLoggers.forEach((logger, level) -> {
+                        if (level.isLessSpecificThan(Level.ERROR)) {
```

Comment:
Hmm actually I'm not sure about this bit. This means we can't override this behaviour at runtime even if we wanted to see these messages. Is that the intention? I guess we have to override `INFO` and `WARN` since these are the possible defaults, but maybe if the level is `DEBUG` or `TRACE` we shouldn't override it?

Reply:
Looking at my change and your comment, I think I should address this differently. What I did here hides the noisy warning messages during `ClientConfiguration` class initialization only. This is still possible that `BasicProfileConfigFileLoader` et al emit more warning messages after that.

I pushed f193c2fabb64e4b94e0d2ee6e3e34fac7b346c74 to change a bit this logic: it now detects if the AWS loggers are explicitly configured in `log4j2.properties` and in case they aren't, their levels are set to `ERROR`. This way we don't override any existing file configuration (if a user really wants to see debug or warn messages) during the startup and until the cluster settings are applied (in which case a different logging level might override the ERROR level we configured). I also added tests.

Let me know what you think.